src/state/routing/utils.ts

```diff
@@ -37,88 +55,113 @@ export function computeRoutes(
-        .map(({ routev2, inputAmount, outputAmount }) => ({ routev2, inputAmount, outputAmount })) ?? [],
+      routes
+        ?.filter(
+          (r): r is typeof routes[0] & { routev2: NonNullable<typeof routes[0]['routev2']> } => r.routev2 !== null
+        )
+        .map(({ routev2, inputAmount, outputAmount }) => ({
+          routev2,
+          inputAmount,
+          outputAmount,
+        })) ?? [],
     v3Routes:
-      route
-        ?.filter((r): r is typeof route[0] & { routev3: NonNullable<typeof route[0]['routev3']> } => r.routev3 !== null)
-        .map(({ routev3, inputAmount, outputAmount }) => ({ routev3, inputAmount, outputAmount })) ?? [],
+      routes
+        ?.filter(
+          (r): r is typeof routes[0] & { routev3: NonNullable<typeof routes[0]['routev3']> } => r.routev3 !== null
+        )
+        .map(({ routev3, inputAmount, outputAmount }) => ({
+          routev3,
+          inputAmount,
+          outputAmount,
+        })) ?? [],
     mixedRoutes:
-      route
+      routes
         ?.filter(
-          (r): r is typeof route[0] & { mixedRoute: NonNullable<typeof route[0]['mixedRoute']> } =>
+          (r): r is typeof routes[0] & { mixedRoute: NonNullable<typeof routes[0]['mixedRoute']> } =>
             r.mixedRoute !== null
         )
-        .map(({ mixedRoute, inputAmount, outputAmount }) => ({ mixedRoute, inputAmount, outputAmount })) ?? [],
+        .map(({ mixedRoute, inputAmount, outputAmount }) => ({
+          mixedRoute,
+          inputAmount,
+          outputAmount,
+        })) ?? [],
     tradeType,
-    gasUseEstimateUSD,
+    gasUseEstimateUSD: parseFloat(gasUseEstimateUSD).toFixed(2).toString(),
     blockNumber,
   })
+
+  return { state: QuoteState.SUCCESS, trade }
 }
 
-const parseToken = ({ address, chainId, decimals, symbol }: GetQuoteResult['route'][0][0]['tokenIn']): Token => {
+function parseToken({ address, chainId, decimals, symbol }: GetQuoteResult['route'][0][0]['tokenIn']): Token {
   return new Token(chainId, address, parseInt(decimals.toString()), symbol)
 }
 
-const parsePool = ({ fee, sqrtRatioX96, liquidity, tickCurrent, tokenIn, tokenOut }: V3PoolInRoute): Pool =>
-  new Pool(
+function parsePool({ fee, sqrtRatioX96, liquidity, tickCurrent, tokenIn, tokenOut }: V3PoolInRoute): Pool {
+  return new Pool(
     parseToken(tokenIn),
     parseToken(tokenOut),
     parseInt(fee) as FeeAmount,
     sqrtRatioX96,
     liquidity,
     parseInt(tickCurrent)
   )
+}
 
 const parsePair = ({ reserve0, reserve1 }: V2PoolInRoute): Pair =>
   new Pair(
     CurrencyAmount.fromRawAmount(parseToken(reserve0.token), reserve0.quotient),
     CurrencyAmount.fromRawAmount(parseToken(reserve1.token), reserve1.quotient)
   )
 
-const genericPoolPairParser = (pool: V3PoolInRoute | V2PoolInRoute): Pool | Pair => {
-  return pool.type === 'v3-pool' ? parsePool(pool) : parsePair(pool)
+export function isExactInput(tradeType: TradeType): boolean {
+  return tradeType === TradeType.EXACT_INPUT
 }
```

Comment:
As a follow-up, this could be moved to a utility and used _everywhere_.

Reply:
the `tradeType === TradeType.EXACT_INPUT` line is in a couple of other places in the codebase - will add a TODO to convert them to using the util